assets/js/private.tsx

```diff
@@ -0,0 +1,69 @@
+import { h, render } from "preact";
+import { useState } from "preact/hooks";
+
+interface NetlifyUser {
+  avatar_url: string;
+  backendName: "github" | string;
+  id: string;
+  login: string; // username
+  name: string; // display name
+  token: string;
+}
+
+const NETLIFY_USER_KEY = "netlify-cms-user";
+
+export function loggedInUser(): NetlifyUser | undefined {
+  const jsonString = localStorage.getItem(NETLIFY_USER_KEY);
+  if (jsonString) {
+    try {
+      const netlifyUser = JSON.parse(jsonString) as NetlifyUser;
+      if (netlifyUser.token) {
+        return netlifyUser;
+      }
+    } catch {}
+  }
+}
+
+function logOut() {
+  localStorage.removeItem(NETLIFY_USER_KEY);
+}
+
+function PrivateLoginButton({ initialUser }: { initialUser?: NetlifyUser }) {
+  const [currentUser, setCurrentUser] = useState(initialUser);
+
+  if (currentUser) {
+    return (
+      <div>
+        <img
+          src={currentUser.avatar_url}
+          alt="Current User Avatar"
+          height="30"
+        />
+        @{currentUser.login}
+        <button
+          type="button"
+          className="usa-button usa-button--gray"
+          onClick={() => {
+            logOut();
+            setCurrentUser(undefined);
+          }}
```

Comment:
It might be nice if the state were more tightly coupled to the localStorage value so that we'd just need to call one function here.

```suggestion
          onClick={logOut}
```

Couple ideas:

- Could move `logOut` inside the component and have it call `setCurrentUser`
- Could create a custom hook to manage localStorage + state, e.g. `const [user, logOut] = useUser();`